src/ResourceManager/IotHub/Commands.IotHub/IotHub/SetAzureRmIotHub.cs

```diff
@@ -122,6 +126,27 @@ public class SetAzureRmIotHub : IotHubBaseCmdlet
         [ValidateNotNullOrEmpty]
         public PSOperationsMonitoringProperties OperationsMonitoringProperties { get; set; }
 
+        [Parameter(
+            ParameterSetName = UpdateRoutingPropertiesParameterSet,
+            Mandatory = false,
+            HelpMessage = "RoutingProperties")]
+        [ValidateNotNullOrEmpty]
+        public PSRoutingProperties RoutingProperties { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateRoutePropertiesParameterSet,
+            Mandatory = false,
+            HelpMessage = "Routes")]
+        [ValidateNotNullOrEmpty]
+        public List<PSRouteProperties> Routes { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateFallbackRoutePropertyParameterSet,
+            Mandatory = false,
+            HelpMessage = "Fallback Routes")]
+        [ValidateNotNullOrEmpty]
+        public PSFallbackRouteProperties FallbackRoute { get; set; }
```

Comment:
Is it really necessary to have 3 different types for these 3 kinds of route properties?